internal/db/schema/migrations/postgres/2/05_job.up.sql

```diff
@@ -0,0 +1,79 @@
+begin;
+
+create table job (
+    id text primary key,
+    name wt_name not null,
+    description wt_description not null,
+    code text not null
+        constraint job_code_must_be_not_empty
+            check(length(trim(code)) > 0),
+    next_scheduled_run timestamp with time zone not null
+        default current_timestamp,
+
+    constraint job_name_code_uq
+        unique(name, code)
+);
+
+comment on table job is
+    'job is a base table where each row represents a unique job that can only have one running instance at any specific time.';
+
+create table job_name_enm (
+    name text not null primary key
+);
+
+comment on table job_name_enm is
+    'job_name_enm is an enumeration table where each row contains the name of a valid job.';
+
+create table job_run (
+     id serial primary key,
+     job_id text not null
+         constraint job_fk
+             references job(id)
+             on delete cascade
+             on update cascade,
+     server_id text
+         constraint server_fk
+             references server(private_id)
+             on delete set null
+             on update cascade,
+     start_time timestamp with time zone not null
+         default current_timestamp,
+     end_time timestamp with time zone,
+     last_heartbeat timestamp with time zone not null
+         default current_timestamp,
+     completed_count int not null,
+     total_count int not null
+         constraint job_run_total_count_greater_than_zero
+            check(total_count > 0),
```

Comment:
Why can't the `total_count` be 0?

Reply:
I was worried about what having a zero value for `total_count` meant and that it could cause div by zero problems.  I was planning to have `total_count` always be at least 1 when the job is started once the job is running and has more information it can always increase this as needed

Reply:
I could easily see a job starting and then completing immediately because there are 0 items to process.

Reply:
This is fair